ide/languages.hcl/src/org/netbeans/modules/languages/hcl/HCLSemanticAnalyzer.java

```diff
@@ -115,9 +121,11 @@ public DefaultHighlighter(SourceRef refs) {
         }
         
         @Override
-        protected boolean visitBlock(HCLBlock block) {
-            if (block.getParent() instanceof HCLDocument) {
-                List<HCLIdentifier> decl = block.getDeclaration();
+        protected void highlight(HCLTreeWalker.Step step) {
+
+            HCLElement<HCLElement> e = step.node();
+            if (e instanceof HCLBlock block && step.depth() == 1) {
```

Comment:
It might be clearer, if the ifs would be chained as `else if`s. Possible double matches would then be prevented.

Reply:
switch with record patterns when the time comes. Till then I've changed them to `else if`-s.